src/java/org/apache/cassandra/net/AsyncStreamingOutputPlus.java

```diff
@@ -177,7 +178,7 @@ public int writeToChannel(Write write, StreamRateLimiter limiter) throws IOExcep
      */
     public long writeFileToChannel(FileChannel file, StreamRateLimiter limiter) throws IOException
     {
-        if (channel.pipeline().get(SslHandler.class) != null)
+        if (channel.pipeline().get(SslHandler.class) != null || !DatabaseDescriptor.getRawConfig().netty_zerocopy_enabled)
```

Comment:
Should we use an actual getter here rather than DatabaseDescriptor.getRawConfig?

Reply:
switched to `DatabaseDescriptor.nettyZerocopyEnabled()`